api_app/script_analyzers/observable_analyzers/firehol_iplist.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import re
+import traceback
+import requests
+import ipaddress
+
+from api_app.exceptions import AnalyzerRunException
+from api_app.script_analyzers import classes
+
+logger = logging.getLogger(__name__)
+
+
+class FireHol_IPList(classes.ObservableAnalyzer):
+    def set_config(self, additional_config_params):
+        self.list_names = additional_config_params.get("list_names", [])
+
+    def run(self):
+        result = {}
+
+        for list_name in self.list_names:
+            result[list_name] = False
+            db = self.download_iplist(list_name)
+
+            if self.observable_name in db:
+                result[list_name] = True
+            else:
+                ip_split = self.observable_name.split(".")
+                subnets = re.findall(
+                    fr"^{ip_split[0]}[\.\d{1,3}\.\d{1,3}\.\d{1,3}/\d{1, 2}]+",
+                    db,
+                    re.MULTILINE,
+                )
+                if len(subnets) > 0:
+                    for subnet in subnets:
+                        if ipaddress.IPv4Address(
+                            self.observable_name
+                        ) in ipaddress.IPv4Network(subnet):
+                            result[list_name] = True
+                            break
+
+        return result
+
+    @staticmethod
+    def download_iplist(list_name):
+        if ".ipset" not in list_name and ".netset" not in list_name:
+            raise AnalyzerRunException(
+                f"extension missing from {list_name} (add .ipset or .netset to name)"
+            )
+
+        try:
+            data_cleaned = ""
+
+            logger.info(f"starting download of {list_name} from firehol iplist")
+            url = f"https://iplists.firehol.org/files/{list_name}"
+            r = requests.get(url)
+            r.raise_for_status()
+
+            data_extracted = r.content.decode()
+
+            for line in data_extracted.splitlines():
+                if not line.startswith("#"):
+                    data_cleaned += f"{line}\n"
```

Comment:
what's the point to use this as a string instead of a list?

Reply:
For the regex search we are doing in `run()`. Am I missing something?

Reply:
the regex is not required, you can just put each element in a list then use ipaddress library to check what they are

Reply:
My point of view was that, if we don't find an IP address directly in the list then we filter out the subnets by matching the first block of given IP and search for the IP under the filtered subnets only. I thought this could reduce the searching we do. So, will there be any performance difference between both the ways? Thank you!

Reply:
which list are you talking about? `if self.observable_name in db:` this is not a check in a list but in a string

You used `ipaddress` library. Please make the effort to understand how to match the IP address by using only this library. There is no need to create huge strings neither to make any kind of regexes.

The only fact that you check whether a string is in a huge strings, instead of iterating in a list and performing a check for each string you saved in a list, is bad for performance. Plus regexes are expensive too. Also we can't  use a custom regex for things (like IP addresses) that follow a very well known schema: we need to leverage the standard library

Reply:
Thank you for explaining it to me. I will change the data into a list and leverage the `ipaddress` library for checking.

`if we don't find an IP address directly in the list`

The list I referred here was fireHol's iplist.